invenio_communities/token.py

```diff
@@ -16,84 +16,59 @@
-                data=extra_data,
-                rnd=binascii.hexlify(os.urandom(4)).decode('utf-8')
-            )
+    Note random data is added to ensure that no two tokens are identical.
+    """
+    jss = JSONWebSignatureSerializer(current_app.config['SECRET_KEY'])
+    return jss.dumps(
+        dict(
+            id=obj_id,
+            data=extra_data,
+            rnd=binascii.hexlify(os.urandom(4)).decode('utf-8')
         )
+    )
 
-    def validate_token(self, token, expected_value=None):
-        """Validate secret link token.
-        :param token: Token value.
-        :param expected_value: A dictionary of key/values that must be present
-            in the data part of the token (i.e. included via ``extra_data`` in
-            ``create_token``).
-        """
-        try:
-            # Load token and remove random data.
-            data = self.load_token(token)
-
-            # Compare expected data with data in token.
-            if expected_value:
-                if expected_value != data["id"]:
-                    return None
-            return data
-        except BadData:
-            return None
-
-    def load_token(self, token, force=False):
-        """Load data in a token.
-        :param token: Token to load.
-        :param force: Load token data even if signature expired.
-                      Default: False.
-        """
-        try:
-            data = self.loads(token)
-        except SignatureExpired as e:
-            if not force:
-                raise
-            data = e.payload
-
-        del data["rnd"]
+
+def validate_token(token, expected_value=None):
+    """Validate secret link token.
+    :param token: Token value.
+    :param expected_value: A dictionary of key/values that must be present
+        in the data part of the token (i.e. included via ``extra_data`` in
+        ``create_token``).
+    """
+    try:
+        # Load token and remove random data.
+        data = load_token(token)
+
+        # Compare expected data with data in token.
+        if expected_value:
+            if expected_value != data["id"]:
+                return None
         return data
+    except BadData:
+        return None
 
 
-class MembershipTokenSerializer(JSONWebSignatureSerializer, TokenMixin):
-    """Serializer for email confirmation link tokens.
-    Depends upon the secrecy of ``SECRET_KEY``. Tokens expire after a specific
-    time (defaults to ``ACCESSREQUESTS_CONFIRMLINK_EXPIRES_IN``). The
-    access request id as well as the email address is stored in the token
-    together with a random bit to ensure all tokens are unique.
+def load_token(token, force=False):
+    """Load data in a token.
+    :param token: Token to load.
+    :param force: Load token data even if signature expired.
+                    Default: False.
     """
+    jss = JSONWebSignatureSerializer(current_app.config['SECRET_KEY'])
+
+    try:
+        data = jss.loads(token)
+    except SignatureExpired as e:
```

Comment:
As discussed IRL, let's instead catch the superclass exception (BadSignature) and re-raise as our own custom HTTPException-based error (to have a proper 400 response returned)